File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        currPenalty, minPenalty, earliestHour = 0, 0, 0
        for i in range(0, len(customers)):
            if customers[i] == 'Y': currPenalty-=1
            else: currPenalty+=1
            if currPenalty < minPenalty:
                minPenalty = currPenalty
                earliestHour = i+1
        return earliestHour

File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.cpp
class Solution {
public:
    int bestClosingTime(string customers) {
        int currPenalty = count(customers.begin(), customers.end(), 'Y');
        int minPenalty = currPenalty, earliestHour = 0;
        for(int i=0; i<customers.size(); i++){
            char c = customers[i];
            if(c == 'Y') currPenalty--;
            else currPenalty++;
            if(currPenalty < minPenalty){
                minPenalty = currPenalty;
                earliestHour = i+1;
            }
        }
        return earliestHour;
    }
};